src/Dap.Protocol/DapReceiver.cs

```diff
@@ -81,16 +83,29 @@ protected virtual IEnumerable<Renor> GetRenor(JToken @object)
                     // This makes it so that the cancel handler implementer must still return a positive response even if the request didn't make it through.
                     if (ro.TryGetValue("requestId", out var requestId))
                     {
-                        yield return new Notification(JsonRpcNames.CancelRequest, JObject.FromObject(new { id = requestId }));
+                        yield return new Notification(JsonRpcNames.CancelRequest, JObject.FromObject(new { id = requestId })){
+                            TraceState = request.TryGetValue("tracestate", out var ts) ? ts.Value<string>() : null,
+                    TraceParent = request.TryGetValue("traceparent", out var tp) ? tp.Value<string>() : null,
+                    };
                         ro.Remove("requestId");
                     }
+                    else
+                    {
+                        yield return new Request(sequence, RequestNames.Cancel, ro) {
+                            TraceState = request.TryGetValue("tracestate", out var ts) ? ts.Value<string>() : null,
+                            TraceParent = request.TryGetValue("traceparent", out var tp) ? tp.Value<string>() : null,
+                        };
+                        yield break;
+                    }
+                }
 
-                    yield return new Request(sequence, RequestNames.Cancel, ro);
+                {
+                    yield return new Request(sequence, requestName, requestObject) {
+                        TraceState = request.TryGetValue("tracestate", out var ts) ? ts.Value<string>() : null,
+                        TraceParent = request.TryGetValue("traceparent", out var tp) ? tp.Value<string>() : null,
```

Comment:
Per [the spec](https://www.w3.org/TR/trace-context/#header-name), this must be a case insensitive header name match. Is that how `request.TryGetValue` is implemented?

Reply:
Thanks!